.github/workflows/validate-projects.yml

```diff
@@ -0,0 +1,78 @@
+name: Validate, Update, and Deploy
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+  schedule:
+    - cron: "45 1 * * *"
+
+jobs:
+  validate-projects-yaml:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.12
+
+      - name: Install dependencies
+        run: pip install -r requirements.txt
+
+      - name: Validate projects YAML file
+        run: python -u tools/check_projects_yaml.py
+
+  test-chert:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.12
+
+      - name: Install dependencies
+        run: pip install -r requirements.txt
+
+      - name: Render site
+        run: chert render
+
+  update-and-deploy:
```

Comment:
I think I see an issue, but it's fixable. Right now it's great that we check the YAML, that's a big improvement. But when we're checking the render, we're still checking it on the non-updated JSON.

So I'm thinking we might want to split the update and commit back? Like, we want to pull the data, and assuming that succeeds, then check render, and if that succeeds, then commit/push the new data. wdyt?

Reply:
One way would be to add a chert render to the `update-and-deploy` job right after the `Generate updated projects.json file` step...

Reply:
Also, a way to improve the usefulness of the `chert render` job would be to turn `reraise` on [here](https://github.com/mahmoud/chert/blob/master/chert/core.py#L777). If this sounds like a good idea, please advise on how to accomplish this. Maybe a separate PR?

Reply:
For now I added the `chert render` step to the `update-and-deploy` job. Let me know if you would prefer a different way.